### ISSUE TITLE
Fatal error in Basic: Can't call method "string" on an undefined value

### ISSUE THREAD

**Issue by pnax** [zonemaster/zonemaster-engine#948]
Testing the domain "kamglass.com.pl" results in the following error:

```
$ zonemaster-cli --test basic kamglass.com.pl                                                                                                           
Seconds Level     Message
======= ========= =======
   1.02 WARNING   Nameserver ns1.webserwer.pl/176.32.162.194 reponds to a SOA query with no SOA records in the answer section.
   1.02 CRITICAL  Fatal error in Basic: Can't call method "string" on an undefined value at ../zonemaster-engine/lib/Zonemaster/Engine/Test/Basic.pm line 673.
```

I attach a file with the data from the test:
[20210809-fatal_error_basic.txt](https://github.com/zonemaster/zonemaster-engine/files/6960362/20210809-fatal_error_basic.txt)

**Comment by matsduf** [comment]
It is interesting to find a name server that gives a truly strange response to a query:
```
; <<>> DiG 9.10.6 <<>> kamglass.com.pl @ns1.webserwer.pl soa +norec
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 40253
;; flags: qr; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;kamglass.com.pl.		IN	SOA

;; Query time: 50 msec
;; SERVER: 176.32.162.194#53(176.32.162.194)
;; WHEN: Tue Aug 31 21:55:40 CEST 2021
;; MSG SIZE  rcvd: 33
```

**Comment by matsduf** [comment]
Resolved by #952.

**Comment by matsduf** [comment]
Resolved by #954 (v2021.1.3) and #952 (v2021.2).
